### DISCUSSION TITLE
Example Unraid Install

### DISCUSSION THREAD

**Discussion by McCloudS** [McCloudS/subgen#137] [Show and tell]
See below for screenshots of what an install would look like.  All you (mostly) need to do is click "Add Container" at the bottom of your Docker page, then fill out the fields to match your settings.  You may need to click "Basic View" in the top right on creation so you can see advanced parameters.  Add appropriate paths/variables that are missing for your desired configuration.

![image](https://github.com/user-attachments/assets/a252b19b-3016-45d3-be68-1d1c7ec7d9a0)
![image](https://github.com/user-attachments/assets/7d0bd1ae-c845-4f21-9d6c-60e5556702dc)
![image](https://github.com/user-attachments/assets/2143e80f-1ab9-41ae-b167-6e9a74ab4497)
![image](https://github.com/user-attachments/assets/8c5e2176-cd93-4ac1-befa-23c48371d4ec)
![image](https://github.com/user-attachments/assets/aaaaa575-ee5e-42cf-86e8-ef00bbb86f8a)

**Comment by CyberBlaed** [comment]
Unraid Template files are stored on the USB in location: /boot/config/plugins/dockerMan/templates-user

You can put it directly on the USB itself, use SMB share, or upload via it's Filesystem in webUI.
I have added all options to the template where possible (selectables easier for the n00b crowd)

Hopefully helps and makes it easier for others.
[my-Subgen.txt](https://github.com/user-attachments/files/17892740/my-Subgen.txt)
- Save as xml file ofcourse, I renamed it as txt to upload it.

(I mean, after upload it can just be a wget command, with output as xml to make it easier as a copy and paste)
----

Only found this today, installed it and got it running, so a big thankyou for that. its wonderful! :D

**Reply by CyberBlaed** [reply]
Copy paste to terminal/command line

`wget -O /boot/config/plugins/dockerMan/templates-user/my-Subgen.xml https://github.com/user-attachments/files/17892740/my-Subgen.txt`

Edit: Helps if i tell it XML for unraid to read it, (rename from txt extention)

**Reply by diogonr** [reply]
Amazing, thank you for the template, it would be great to add to the main repository. 
Minor note for people doing this without Nvidia GPUs:
- Click advanced view.
- On Extra Parameters, remove "--gpus all".

**Comment by jcouch93** [comment]
Thank you for the write-up! Everything works, except I can't get the Bazarr instance to connect to the Subgen instance. It gives me a connection refused error.

Also, I have some questions about what is happening? I see:
2025-03-02 18:40:20,483 DEBUG: No subtitles in 'Unknown' language found in the video.
2025-03-02 18:40:20,483 DEBUG: Searching for subtitles in: /data/media/Couch_TV_Library/Walker, Texas Ranger (1993)/Season 08
2025-03-02 18:40:20,492 DEBUG: Skipping subtitles because language is NONE.
2025-03-02 18:40:20,492 INFO: Skipping Walker, Texas Ranger (1993) - S08E08 - Widow Maker (DVD) (XviD).avi: Subtitles already exist in None.

I know this file has an external subtitle file, but I'm curious about the "None" string above. Do you have more info?

If the video file does not have a subtitle file, would one then get created?

Thanks!

**Reply by McCloudS** [reply]
Are you ultimately trying to use Bazarr or not?  It looks like you added directories.  What IP address did you use as your provider?  It should be your unraid server IP.  

To get around the NONE part, it's because your files don't appear to have a audio track set to a language.  `SHOULD_WHISPER_DETECT_AUDIO_LANGUAGE` to True should get you further.  I also found a flaw with some detection, so `UPDATE=TRUE` should also fix your issue.  I ultimately need to set aside some time to re-write some of the logic so it's more clear.

**Reply by jcouch93** [reply]
I was originally trying to use Bazarr, but gave up after I couldn't get the connection to test good. So if I use only Bazarr, I don't need to input my folders either? The folders are only needed for a "standalone" deployment?

Does the Nvidia GPU require the --runtimenvidia and the Variable Nvidia Visible Devices? I added them but got a incompatible hardware error, (CUDA failed with error forward compatibility was attempted on non supported HW) but I think that may be due to I need to update the driver.

**Reply by McCloudS** [reply]
> I was originally trying to use Bazarr, but gave up after I couldn't get the connection to test good. So if I use only Bazarr, I don't need to input my folders either? The folders are only needed for a "standalone" deployment?

Correct and correct.  Did you figure out Bazarr?
> Does the Nvidia GPU require the --runtimenvidia and the Variable Nvidia Visible Devices? I added them but got a incompatible hardware error, (CUDA failed with error forward compatibility was attempted on non supported HW) but I think that may be due to I need to update the driver.

I only have `--gpus all`
What is your GPU?

**Reply by jcouch93** [reply]
GPU is a 1660TI.I'm running everything in unraid dockers, so the difficulty is upped a bit. I can ping from the Bazarr container to the Subgen container, but I get an error when I perform the test in Bazarr (error 111 connection refused). I don't know why it can't connect as as far as I can tell the mapping is correct. Still trying to figure that out but I thought I would just use as standalone until I can figure that out.

**Reply by McCloudS** [reply]
If you didn’t change any of the template settings, it will just be http://{yourunraidserverip}:9000

The container is configured for host mode with a default port of 9000.

**Reply by jcouch93** [reply]
I had the container in Bridge mode, so after your last message I changed it to Host mode and it now connects.

Next on the list is getting the GPU working. Thanks for all your help so far!

**Reply by McCloudS** [reply]
You have the nvidia driver plugin installed?

**Reply by jcouch93** [reply]
I do. I got it working after updating to 6.12.15 and updating the driver. I removed the video folders from the app and integrated it into Bazarr, and it appears the integration is successful as I can see it processing the file and see it added to the folder. Do you know if there's a way to get better separation when two voices are talking? Or more pauses when two different people are speaking? IF not no big deal. Thank you again for all your help!

**Reply by McCloudS** [reply]
> I do. I got it working after updating to 6.12.15 and updating the driver. I removed the video folders from the app and integrated it into Bazarr, and it appears the integration is successful as I can see it processing the file and see it added to the folder. Do you know if there's a way to get better separation when two voices are talking? Or more pauses when two different people are speaking? IF not no big deal. Thank you again for all your help!

Not really any way to get that at this time (with Subgen).  https://github.com/ahmetoner/whisper-asr-webservice has whisperX which has diariziation, which may be closer to what you want.

**Reply by buzdriver** [reply]
Hi,
can you run it with CLI? 
Example: docker exec "$DOCKER_NAME" subgen -input $name.$inputlang.srt -output $name.$outputlang.srt
/BR

**Reply by McCloudS** [reply]
> Hi, can you run it with CLI? Example: docker exec "$DOCKER_NAME" subgen -input $name.$inputlang.srt -output $name.$outputlang.srt /BR

Nope.  At that rate you could run Stable-TS or Whisper.

**Comment by loonsies** [comment]
Isn't the config you get by using the command above missing SKIP_IF_EXTERNAL_SUBTITLES_EXIST ?

Edit: well it seemed outdated anyway. Here's an updated one: 

https://gist.github.com/loonsies/e8684b0c1bd0f11a86566b916975579f

**Reply by BraveCaptain** [reply]
bravo
